hypershift-operator/controllers/nodepool/metrics/metrics.go

```diff
@@ -31,8 +31,8 @@ const (
 	CountByHClusterMetricName = "hypershift_hostedcluster_nodepools" // What about renaming it to hypershift_cluster_nodepools ?
 	countByHClusterMetricHelp = "Number of NodePools for a given HostedCluster"
 
-	CoresCountByHClusterMetricName = "hypershift_cluster_cores"
-	CoresCountByHClusterMetricHelp = "Number of cores for a given HostedCluster. " +
+	VCpusCountByHClusterMetricName = "hypershift_cluster_vcpus"
+	VCpusCountByHClusterMetricHelp = "Number of vCpus for a given HostedCluster. " +
 		"-1 if this number cannot be computed." // Be careful when changing this metric as it is used for billing the customers
```

Comment:
IIUC "cannot be computed" could have different causes:
* The operator doesn't know how to retrieve the information from the platform (e.g. anything else than AWS).
* The operator can't find information for the EC2 instance type of the pool.
* else?

It might be helpful to report what caused the issue, maybe via another metric.

Reply:
I'm more a metric person :)
My experience with logs is that nobody will look at them while a metric will be collected in all cases.

Reply:
For context, `If you can log it, you can have a metric for it` from  https://the-zen-of-prometheus.netlify.app/